internal/controlplane/handlers_entities.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+
+	"github.com/stacklok/minder/internal/db"
+	"github.com/stacklok/minder/internal/engine"
+	"github.com/stacklok/minder/internal/events"
+	"github.com/stacklok/minder/internal/logger"
+	"github.com/stacklok/minder/internal/providers"
+	"github.com/stacklok/minder/internal/reconcilers/messages"
+	"github.com/stacklok/minder/internal/util"
+	pb "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+// ReconcileEntityRegistration reconciles the registration of an entity.
+//
+// Currently, this method only supports repositories but is intended to be
+// generic and handle all types of entities.
+// Todo: Utilise for other entities when such are supported.
+func (s *Server) ReconcileEntityRegistration(
+	ctx context.Context,
+	in *pb.ReconcileEntityRegistrationRequest,
+) (*pb.ReconcileEntityRegistrationResponse, error) {
+	l := zerolog.Ctx(ctx).With().Logger()
+
+	entityCtx := engine.EntityFromContext(ctx)
+	projectID := entityCtx.Project.ID
+
+	logger.BusinessRecord(ctx).Project = projectID
+
+	// Todo: We don't support other entities yet. This should be updated when we do.
+	entityType := in.GetEntity()
+	if pb.EntityFromString(entityType) != pb.Entity_ENTITY_REPOSITORIES {
+		return nil, util.UserVisibleError(codes.Internal, "entity type %s not supported", entityType)
```

Comment:
One small change please, this should not return `codes.Internal`, but `codes.InvalidArgument` instead. Mainly so that we get the proper HTTP error code through the gateway (now we get a 500)

Reply:
Yep, the copy/pasta betrayed me, thanks! 
Fixed